src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -272,7 +276,8 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
 
             foreach (var formatter in formatters)
             {
-                if (formatter.CanWriteResult(formatterContext, contentType: null))
+                formatterContext.ContentType = null;
```

Comment:
The bug would be in a user-implemented formatter at that point. I don't think it's sensible to throw or assert here when we can just do the right thing always.